microceph/cmd/microceph/cluster_bootstrap.go

```diff
@@ -33,18 +39,56 @@ func (c *cmdClusterBootstrap) Run(cmd *cobra.Command, args []string) error {
 
 	m, err := microcluster.App(context.Background(), microcluster.Args{StateDir: c.common.FlagStateDir, Verbose: c.common.FlagLogVerbose, Debug: c.common.FlagLogDebug})
 	if err != nil {
-		return fmt.Errorf("Unable to configure MicroCeph: %w", err)
+		return fmt.Errorf("unable to configure MicroCeph: %w", err)
 	}
 
 	// Get system hostname.
 	hostname, err := os.Hostname()
 	if err != nil {
-		return fmt.Errorf("Failed to retrieve system hostname: %w", err)
+		return fmt.Errorf("failed to retrieve system hostname: %w", err)
 	}
 
-	// Get system address.
+	// Get system address for microcluster bootstrap.
 	address := util.NetworkInterfaceAddress()
-	address = util.CanonicalNetworkAddress(address, 7443)
+	address = util.CanonicalNetworkAddress(address, common.BootstrapPortConst)
+
+	// Get parameter data for Ceph bootstrap.
+	var e error
+	var data types.Bootstrap
+	if len(c.flagMonIp) > 0 {
+		data, e = getCephBootstrapData(c.flagMonIp)
+	} else {
+		data, e = getCephBootstrapData(util.NetworkInterfaceAddress())
+	}
+	if e != nil {
+		return fmt.Errorf("failed to parse bootstrap data: %w", err)
+	}
+
+	// Bootstrap microceph daemon.
+	err = m.NewCluster(hostname, address, time.Second*30)
+	if err != nil {
+		return err
+	}
+
+	// TODO: Check if polling is required here.
```

Comment:
Do we or does the client retry do this for us?

Reply:
Added retry.